src/lib/y2storage/proposal/autoinst_drive_planner.rb

```diff
@@ -64,11 +64,60 @@ def planned_devices(_drive)
       #   section containing the partition one
       def configure_device(device, partition_section, drive_section)
         configure_filesystem(device, partition_section, drive_section)
+        configure_usage(device, partition_section)
         add_encryption_attrs(device, partition_section)
       end
 
       alias_method :device_config, :configure_device
 
+      # Sets those attributes that determine how the device will be used (file system, RAID member, etc.)
+      #
+      # A device can be used as file system, a RAID member, an LVM PV, a Bcache backing/caching device
+      # or as part of a Btrfs multi-device. And in the future, the list might grow.
+      #
+      # The logic to determine which device to honor is quite simple and it could be improved
+      # in the future if needed.
+      #
+      # @param device  [Planned::Device] Planned device
+      # @param partition_section [AutoinstProfile::PartitionSection] AutoYaST
+      #   specification of the concrete device
+      def configure_usage(device, partition_section)
+        usage_attr, *ignored_attrs = usage_attrs_in(partition_section)
+        return if usage_attr.nil?
+
+        meth = USAGE_ATTRS_MAP[usage_attr] || usage_attr
+        meth = "#{meth}="
+        device.public_send(meth, partition_section.public_send(usage_attr)) if device.respond_to?(meth)
+        return if ignored_attrs.empty?
+
+        issues_list.add(:conflicting_attrs, partition_section, usage_attr, ignored_attrs)
+      end
+
+      # @return [Array<Symbol>] List of 'usage' attributes. The list is ordered by precedence.
+      USAGE_ATTRS = [
+        :mount, :raid_name, :lvm_group, :btrfs_name, :bcache_backing_for, :bcache_caching_for
+      ]
+      private_constant :USAGE_ATTRS
+
+      # @return [Array<Symbol>] Map of 'usage' attributes to planned device methods. The map
+      #   only contains those attributes which names does not match.
+      USAGE_ATTRS_MAP = {
+        lvm_group: :lvm_volume_group_name,
+        mount:     :mount_point
+      }
+      private_constant :USAGE_ATTRS_MAP
+
+      # Returns the attributes which determines how the device will be used
+      #
+      # @param partition_section [AutoinstProfile::PartitionSection] Partition specification
+      # @return [Array<Symbol>] List of usage related attributes which are defined
+      def usage_attrs_in(partition_section)
+        USAGE_ATTRS.reject do |e|
+          value = partition_section.public_send(e)
+          value.nil? || (value.is_a?(Array) && value.empty?)
```

Comment:
so empty hash does not make sense? If makes, then maybe respond_to?(:empty) and call it can make more sense.

Reply:
No, it does not make sense. They can only be arrays and scalar values.